src/main/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,77 @@
+package calculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class StringCalculator {
+    private static final String CUSTOM_WITH_DIGITAL_REGEX = "(.*)\\d(.*)";
+
+    public static Integer splitAndSum(String input) {
+        if (input == null || input.isEmpty()) {
+            return 0;
+        }
+        return sumList(convertToIntegerArray(splitString(input)));
+    }
+
+    public static String[] splitString(String input) {
+        String regExp = ",|:";
+        if (isCustom(input)) {
+            int idx = input.indexOf("\n");
+            regExp = makeCustom(input, idx);
+            input = input.substring(idx + 1);
+        }
+
+        return input.split(regExp);
+    }
+
+    private static String makeCustom(String input, int idx) {
+        String custom = input.substring(2, idx);
+        if (custom.matches(CUSTOM_WITH_DIGITAL_REGEX)) {
+            throw new RuntimeException("커스텀 구분자 내에 숫자가 포함되어 있습니다.");
+        }
+        return custom;
+    }
+
+    private static boolean isCustom(String input) {
+        return input.startsWith("//") && input.contains("\n");
+    }
+
+    public static List<Integer> convertToIntegerArray(String[] strArr) {
+        List<Integer> numList = new ArrayList<>();
+        if (isEmptyList(strArr)) {
+            return numList;
+        }
+
+        for (String num : strArr) {
+            numList.add(convertToInt(num));
+        }
+        return numList;
+    }
+
+    private static boolean isEmptyList(String[] strArr) {
+        return Arrays.stream(strArr).count() == 1 && strArr[0].equals("");
+    }
+
+    private static Integer convertToInt(String str) {
+        int value;
+        try {
+            value = Integer.parseInt(str);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("입력 값은 숫자여야 합니다.");
+        }
+        isPositive(value);
+
+        return value;
+    }
+
+    private static void isPositive(int value) {
+        if (value < 0) {
+            throw new RuntimeException("입력 값은 0 이상의 정수여야 합니다.");
+        }
+    }
+
+    public static Integer sumList(List<Integer> numList) {
+        return numList.stream().mapToInt(Integer::intValue).sum();
```

Comment:
한줄에 점 하나씩이라는 개념으로 스트림을 한줄에 하나의 메서드로 처리한다면, 더 가독성은 높아질거 같습니다.
```suggestion
        return numList.stream()
                .mapToInt(Integer::intValue)
                .sum();
```
아울러 아래 사진과 같이 IDE에서 각 연산마다 어떤 값이 return되는지도 확인할 수 있어 더 편리한거 같습니다.

![image](https://user-images.githubusercontent.com/53366407/153602667-bd2ca559-90e1-429d-8021-3d6e7154486a.png)

Reply:
해당 방식을 이용하면 더욱 직관적일 것 같습니다!! 다만, 제 IDEA 안에서는
![image](https://user-images.githubusercontent.com/67397679/153824747-2f8e8bbf-3690-41f2-a6fa-ede9a875e68d.png)
중간 연산 값들이 표시가 안되고 있는데, 해결 방법이 있을까요?